File: app/src/main/java/com/constraint/vagabond/main/details/DetailsActivity.java
package com.constraint.vagabond.main.details;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.constraint.vagabond.R;
import com.constraint.vagabond.data.DataStore;
import com.constraint.vagabond.data.RecreationalAreaList;
import com.google.android.material.appbar.CollapsingToolbarLayout;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.squareup.picasso.Picasso;

import java.util.List;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class DetailsActivity extends AppCompatActivity implements DetailsContract.View{

  private CollapsingToolbarLayout collapsingToolbarLayout;
  private Toolbar toolbar;
  private FloatingActionButton favButton;
  private String collapsingImageUrl;
  private TextView areaDescription;
  private RecreationalAreaList detailedArea = DataStore.getRecreationalAreaList();
  private int position;
  private RecyclerView areaImages;
  private Button directionBtn;

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    Intent getData = getIntent();
    collapsingImageUrl = getData.getStringExtra("image_url");
    position = getData.getIntExtra("position", 0);
//    imageUrls = detailedArea.recreationalAreaList.get(position).getImageUrls();
    initializeView();
  }

  @Override
  protected void onStart() {
    super.onStart();
    initializeData();
    loadImage();
//    setDataToRecyclerView();
  }

  public void initializeView() {
    setContentView(R.layout.activity_detail);
    toolbar = findViewById(R.id.toolbar_title);
    setSupportActionBar(toolbar);
    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    collapsingToolbarLayout = findViewById(R.id.collapsing_layout);
    areaDescription = findViewById(R.id.description);
    favButton = findViewById(R.id.fav_button);
    areaImages = findViewById(R.id.horizontal_recycler_view);
    areaDescription = findViewById(R.id.description);
    directionBtn = findViewById(R.id.directions);
  }


  public void loadImage() {
    final ImageView expandingImage = findViewById(R.id.collapsing_image);
    Picasso.get().load(collapsingImageUrl).fit().centerCrop().into(expandingImage);
  }

  @Override
  public boolean onCreateOptionsMenu(Menu menu) {
    MenuInflater inflator = getMenuInflater();
    inflator.inflate(R.menu.menu_main, menu);
    return true;
  }

  @Override
  public void onBackPressed() {
    super.onBackPressed();
    finish();
  }

  public void initializeData() {
    collapsingToolbarLayout.setTitle(detailedArea.recreationalAreaList.get(position).recAreaName);
    String desc = detailedArea.recreationalAreaList.get(position).recAreaDescription;
    areaDescription.setText(desc);
  }

  @Override
  public void setDataToRecyclerView(List<String> imageUrls) {
    LinearLayoutManager linearLayoutManager =
        new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);
    areaImages.setLayoutManager(linearLayoutManager);
    DetailsPhotosAdapter detailsPhotosAdapter = new DetailsPhotosAdapter(imageUrls);
    areaImages.setAdapter(detailsPhotosAdapter);
  }

}

File: app/src/main/java/com/sierra/vagabond/main/details/DetailsActivity.kt
package com.sierra.vagabond.main.details

import android.os.Bundle
import android.view.Menu
import android.widget.ImageView
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.sierra.vagabond.R
import com.sierra.vagabond.data.entities.RecreationalArea
import com.sierra.vagabond.main.details.adapter.DetailsPhotosAdapter
import com.sierra.vagabond.main.details.adapter.FacilitiesAdapter
import com.sierra.vagabond.utils.REC_AREA_ID
import com.sierra.vagabond.viewmodels.DetailsActivityViewModel
import com.sierra.vagabond.viewmodels.DetailsViewModelFactory
import com.sierra.vagabond.viewmodels.MainViewModelFactory
import com.squareup.picasso.Picasso
import dagger.android.AndroidInjection
import kotlinx.android.synthetic.main.activity_detail.*
import javax.inject.Inject

class DetailsActivity : AppCompatActivity(), DetailsMvp.View {

    @Inject
    lateinit var viewModelFactory: DetailsViewModelFactory

    private val detailsViewModel: DetailsActivityViewModel by viewModels { viewModelFactory }

    private lateinit var id: String

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        id = intent.getStringExtra(REC_AREA_ID)
        setSupportActionBar(toolbar_title)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        detailsViewModel.area.observe(this, Observer { area ->
            initializeData(area)
        })
        detailsViewModel.getSingleArea(id)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }

    private fun initializeData(detailedArea: RecreationalArea) {
        val expandingImage = findViewById<ImageView>(R.id.collapsing_image)
        Picasso.get().load(detailedArea.recAreaMediaList[0].imageURL).fit().centerCrop().into(expandingImage)
        collapsing_layout.title = detailedArea.recAreaName
        val desc = detailedArea.recAreaDescription
        description.text = desc
        val linearLayoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        list_facilities.layoutManager = linearLayoutManager
        val facilities = detailedArea.recAreaFacilities
        val facilitiesAdapter = FacilitiesAdapter(facilities, detailsViewModel)
        list_facilities.adapter = facilitiesAdapter
    }

    override fun setDataToRecyclerView(imageUrls: List<String>) {
        val linearLayoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        horizontal_recycler_view.layoutManager = linearLayoutManager
        val detailsPhotosAdapter = DetailsPhotosAdapter(imageUrls)
        horizontal_recycler_view.adapter = detailsPhotosAdapter
    }
}
